File: dx-api/app/constants/game_pattern.go
package constants

// Game pattern values.
const (
	GamePatternListen = "listen"
	GamePatternSpeak  = "speak"
	GamePatternRead   = "read"
	GamePatternWrite  = "write"
)

// DefaultGamePattern is the default game pattern.
const DefaultGamePattern = GamePatternWrite

// GamePatternLabels maps each pattern to its Chinese label.
var GamePatternLabels = map[string]string{
	GamePatternListen: "听",
	GamePatternSpeak:  "说",
	GamePatternRead:   "读",
	GamePatternWrite:  "写",
}

File: dx-web/src/consts/game-pattern.ts
export const GAME_PATTERNS = {
  LISTEN: "listen",
  SPEAK: "speak",
  READ: "read",
  WRITE: "write",
} as const;

export type GamePattern = (typeof GAME_PATTERNS)[keyof typeof GAME_PATTERNS];

export const GAME_PATTERN_LABELS: Record<GamePattern, string> = {
  listen: "听",
  speak: "说",
  read: "读",
  write: "写",
};

export const DEFAULT_GAME_PATTERN: GamePattern = GAME_PATTERNS.WRITE;
